app/models/user.rb

```diff
@@ -25,6 +25,7 @@ class User < ApplicationRecord
   attribute :password
 
   after_save :ensure_default_role
+  after_initialize :set_default_language_for_new_user, if: :new_record?
```

Comment:
By the way Rails 5 brought us:
```suggestion
  attribute :locale, :string, default: -> { Setting[:default_language].presence }
```
Haven't used it yet, do not know the pitfalls, but I think it is worth trying out, as it looks much cleaner :wink:

Reply:
will try. thanks for your suggestions.